src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -1062,6 +1064,22 @@ private bool CanToggleIsFrozen(object parameters)
             return DynamoSelection.Instance.Selection.Count() == 1;
         }
 
+        /// <summary>
+        /// When a node is frozen, raise the IsFrozen property changed event on
+        /// all its downstream nodes, to ensure UI updates correctly.
+        /// </summary>
+        private void RaisePropertyChangedOnDownStreamNodes()
```

Comment:
Frozen is not directly set on the downstream nodes. So, raise the property changed event on downstream nodes. 
This is raised only when the user explicitly freeze a node. So there will not be any performance change.